src/ast/transform/ExpandEqrels.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2020, The Souffle Developers. All rights reserved.
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file ExpandEqrels.cpp
+ *
+ ***********************************************************************/
+
+#include "ast/transform/ExpandEqrels.h"
+#include "AggregateOp.h"
+#include "RelationTag.h"
+#include "ast/Aggregator.h"
+#include "ast/Argument.h"
+#include "ast/Atom.h"
+#include "ast/BinaryConstraint.h"
+#include "ast/BooleanConstraint.h"
+#include "ast/Clause.h"
+#include "ast/Constant.h"
+#include "ast/Constraint.h"
+#include "ast/Functor.h"
+#include "ast/IntrinsicFunctor.h"
+#include "ast/Literal.h"
+#include "ast/Negation.h"
+#include "ast/Node.h"
+#include "ast/NumericConstant.h"
+#include "ast/Program.h"
+#include "ast/QualifiedName.h"
+#include "ast/RecordInit.h"
+#include "ast/Relation.h"
+#include "ast/TranslationUnit.h"
+#include "ast/TypeCast.h"
+#include "ast/UnnamedVariable.h"
+#include "ast/UserDefinedFunctor.h"
+#include "ast/Variable.h"
+#include "ast/analysis/PolymorphicObjects.h"
+#include "ast/utility/NodeMapper.h"
+#include "ast/utility/Utils.h"
+#include "ast/utility/Visitor.h"
+#include "souffle/BinaryConstraintOps.h"
+#include "souffle/utility/MiscUtil.h"
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <ostream>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace souffle::ast::transform {
+
+bool ExpandEqrelsTransformer::transform(TranslationUnit& translationUnit) {
+    bool changed = false;
+    Program& program = translationUnit.getProgram();
+
+    for (auto* relation : program.getRelations()) {
+        // Only concerned with eqrel relations
+        if (relation->getRepresentation() != RelationRepresentation::EQREL) {
+            continue;
+        }
+        const auto& relName = relation->getQualifiedName();
+        changed = true;
+
+        // Remove eqrel representation
+        relation->setRepresentation(RelationRepresentation::BTREE);
+
+        // (1) Transitivity
+        // A(x,z) :- A(x,y), A(y,z).
+        auto transitiveClause = mk<Clause>();
```

Comment:
This will no longer work after merging my code change.  Head is always required.  I suggest 

```suggestion
        auto transitiveClause = mk<Clause>(relName);
        auto transitiveClauseHead = transitiveClause->getHead();
```

Reply:
No problem, I will make these changes. Will lead to better codeflow, I think.